### PR TITLE
[PM-11200] Move delete item permission to Can Manage

### PR DESCRIPTION
**Pull request by gbubemismith**
## 🎟️ Tracking
[PM-11200](https://bitwarden.atlassian.net/browse/PM-11200)

## 📔 Objective
Currently, users with Can Edit permissions can delete items within the collection. This should be updated so that only users with the Can Manage permission can delete items within a collection.

## 📸 Screenshots

https://github.com/user-attachments/assets/86ac1abd-7b2e-43e6-ac72-899533b4118a

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-11200]: https://bitwarden.atlassian.net/browse/PM-11200?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by shane-melton — CHANGES_REQUESTED** [review]
This looks good for the Individual Vault, but I suspect we're going to run into an issue with the AC/org vault that we'll want to address in the share vault items component.

Also, just an fyi, this is likely going to conflict with https://github.com/bitwarden/clients/pull/10784

**Comment by gbubemismith** [comment]
> This looks good for the Individual Vault, but I suspect we're going to run into an issue with the AC/org vault that we'll want to address in the shared vault items component.
> 
> Also, just an fyi, this is likely going to conflict with #10784

Thanks for pointing this out.I didn't even think of AC

File: apps/web/src/app/vault/components/vault-items/vault-items.component.ts [hunk 1]
```diff
@@ -224,12 +224,22 @@ export class VaultItemsComponent {
       return true;
     }
 
+    // Check for admin access in AC vault
+    if (this.showAdminActions) {
+      const organization = this.allOrganizations.find((o) => o.id === cipher.organizationId);
+
+      if (organization?.allowAdminAccessToAllCollectionItems) {
```
[the remark this answers is not in the document]
Comment (gbubemismith) [reply apps/web/src/app/vault/components/vault-items/vault-items.component.ts]: Thanks that makes sense

File: apps/web/src/app/vault/components/vault-items/vault-items.component.ts [hunk 2]
```diff
@@ -228,7 +228,10 @@ export class VaultItemsComponent {
     if (this.showAdminActions) {
       const organization = this.allOrganizations.find((o) => o.id === cipher.organizationId);
 
-      if (organization?.allowAdminAccessToAllCollectionItems) {
+      if (
+        organization?.allowAdminAccessToAllCollectionItems &&
+        (organization.isAdmin || organization.permissions.editAnyCollection)
```
Comment (shane-melton) [inline apps/web/src/app/vault/components/vault-items/vault-items.component.ts:233]: ⚠️ `editAnyCollection` is not restricted by `allowAdminAccessToAllCollectionItems`. We should pull that condition to its own `if` statement or separate `||` condition.

**Review by shane-melton — CHANGES_REQUESTED** [review]

**Pull request merged**